pkg/engine/artifacts.go

```diff
@@ -33,120 +33,119 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
-			sourceFile:      "kubernetesmasteraddons-omsagent-daemonset.yaml",
+			sourceFile:      containerMonitoringAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.ContainerMonitoringAddonName),
-			destinationFile: "omsagent-daemonset.yaml",
+			destinationFile: containerMonitoringAddonDestinationFilename,
 		},
 		common.IPMASQAgentAddonName: {
-			sourceFile:      "ip-masq-agent.yaml",
+			sourceFile:      ipMasqAgentAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.IPMASQAgentAddonName),
-			destinationFile: "ip-masq-agent.yaml",
+			destinationFile: ipMasqAgentAddonDestinationFilename,
 		},
 		common.AzureCNINetworkMonitorAddonName: {
-			sourceFile:      "azure-cni-networkmonitor.yaml",
+			sourceFile:      azureCNINetworkMonitorAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.AzureCNINetworkMonitorAddonName),
-			destinationFile: "azure-cni-networkmonitor.yaml",
+			destinationFile: azureCNINetworkMonitorAddonDestinationFilename,
 		},
 		common.DNSAutoscalerAddonName: {
-			sourceFile:      "dns-autoscaler.yaml",
+			sourceFile:      dnsAutoscalerAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.DNSAutoscalerAddonName),
-			destinationFile: "dns-autoscaler.yaml",
+			destinationFile: dnsAutoscalerAddonDestinationFilename,
 		},
 		common.CalicoAddonName: {
-			sourceFile:      "kubernetesmasteraddons-calico-daemonset.yaml",
+			sourceFile:      calicoAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.CalicoAddonName),
-			destinationFile: "calico-daemonset.yaml",
+			destinationFile: calicoAddonDestinationFilename,
 		},
 		common.AzureNetworkPolicyAddonName: {
-			sourceFile:      "kubernetesmasteraddons-azure-npm-daemonset.yaml",
+			sourceFile:      azureNetworkPolicyAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.AzureNetworkPolicyAddonName),
-			destinationFile: "azure-npm-daemonset.yaml",
+			destinationFile: azureNetworkPolicyAddonDestinationFilename,
 		},
 		common.AzurePolicyAddonName: {
-			sourceFile:      "azure-policy-deployment.yaml",
+			sourceFile:      azurePolicyAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.AzurePolicyAddonName),
-			destinationFile: "azure-policy-deployment.yaml",
+			destinationFile: azurePolicyAddonDestinationFilename,
 		},
 		common.CloudNodeManagerAddonName: {
-			sourceFile:      "kubernetesmasteraddons-cloud-node-manager.yaml",
+			sourceFile:      cloudNodeManagerAddonSourceFilename,
 			base64Data:      "",
-			destinationFile: "cloud-node-manager.yaml",
+			destinationFile: cloudNodeManagerAddonDestinationFilename,
 		},
 		common.NodeProblemDetectorAddonName: {
-			sourceFile:      "node-problem-detector.yaml",
+			sourceFile:      nodeProblemDetectorAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.NodeProblemDetectorAddonName),
-			destinationFile: "node-problem-detector.yaml",
-			isEnabled:       k.IsAddonEnabled(common.NodeProblemDetectorAddonName),
```

Comment:
Cleaning this up. This addon was introduced while this PR was in development: https://github.com/Azure/aks-engine/pull/2381